### ISSUE TITLE
Solex Desktop and Solex cc sdk

### ISSUE THREAD

**Issue by zeev-mindali** [solexapp/solex-desktop-public#5]
Dear sir,

is there any eta for Solex desktop?
where I can get Solex cc sdk?

zeev

**Comment by kellyschrock** [comment]
Hi Zeev,

I'm pretty close on the Solex Desktop app. I don't have MP4 video working yet, but the missions and flight control functionality is complete. 

If you're interested in SolexCC, I can help you get set up with that. 

Thanks,
Kelly

**Comment by zeev-mindali** [comment]
what is the problem with mp4, maybe I can assist you.
I using today Herelink w/solex, looking for desktop solution.
I am full stuck developer (java) and android iPhone developer, so feel free to ask me anything :)

**Comment by zeev-mindali** [comment]
and yes, I would like to install the soles cc sdk
to play with it

**Comment by zeev-mindali** [comment]
dear friend, any progress.
I also would like to test the soles cc sdk, waiting for your instruction

**Comment by kellyschrock** [comment]
Hi Zeev,

Let me know your github id and I'll invite you to the solexCC repo if you
want to take a look.

**Comment by zeev-mindali** [comment]
https://github.com/zeev-mindali

happy to assist

**Comment by zeev-mindali** [comment]
{
  "login": "zeev-mindali",
  "id": 23069323,
  "node_id": "MDQ6VXNlcjIzMDY5MzIz",
  "avatar_url": "https://avatars.githubusercontent.com/u/23069323?v=4",
  "gravatar_id": "",
  "url": "https://api.github.com/users/zeev-mindali",
  "html_url": "https://github.com/zeev-mindali",
  "followers_url": "https://api.github.com/users/zeev-mindali/followers",
  "following_url": "https://api.github.com/users/zeev-mindali/following{/other_user}",
  "gists_url": "https://api.github.com/users/zeev-mindali/gists{/gist_id}",
  "starred_url": "https://api.github.com/users/zeev-mindali/starred{/owner}{/repo}",
  "subscriptions_url": "https://api.github.com/users/zeev-mindali/subscriptions",
  "organizations_url": "https://api.github.com/users/zeev-mindali/orgs",
  "repos_url": "https://api.github.com/users/zeev-mindali/repos",
  "events_url": "https://api.github.com/users/zeev-mindali/events{/privacy}",
  "received_events_url": "https://api.github.com/users/zeev-mindali/received_events",
  "type": "User",
  "site_admin": false,
  "name": "zeev mindali",
  "company": "amm sec",
  "blog": "",
  "location": null,
  "email": null,
  "hireable": null,
  "bio": null,
  "twitter_username": null,
  "public_repos": 13,
  "public_gists": 0,
  "followers": 3,
  "following": 1,
  "created_at": "2016-10-26T04:47:00Z",
  "updated_at": "2021-10-19T17:10:23Z"
}

**Comment by kellyschrock** [comment]
Cool, sent you an invite.

**Comment by zeev-mindali** [comment]
dear friend, got the invitation for solex-cc :)
but I was also would like to see the desktop if it's possible.


zeev

**Comment by kellyschrock** [comment]
Ah, well the desktop app is currently private. I may open-source it at some
point, but I haven't decided to yet.

Thanks,
Kelly

**Comment by zeev-mindali** [comment]
Understood,

but if you want to take it to next level (enterprise solution) and make joint venture, write to me to my email, so we can talk privately

**Comment by zeev-mindali** [comment]
if you sell it, I would like also. to hear

**Comment by kellyschrock** [comment]
Understood. You have my email address. Please send an email and explain to
me the precise meaning of "next level" and how the joint venture you
describe would work.

**Comment by joshbowyer** [comment]
Have you decided on opensourcing the desktop app? Your website states that it will be compatible with Linux; not sure how you would charge for that but Id be ok paying you directly for personal download/use.

**Comment by kellyschrock** [comment]
Hi @joshbowyer, I'm still working on Solex/Desktop, finishing up a few details. I have a Windows build of it working, and just added (very) basic support for streaming video from the vehicle into a separate gstreamer viewer window. At some point soon I'll get video into the app window itself, but I'm a bit pressed for time ATM. If you're looking for a Linux package, I can get one to you. Are you on a system that can install from .deb files?

**Comment by joshbowyer** [comment]
Yes, I am on Debian 11. Thank you!

**Comment by melanie-donne** [comment]
Hi, it is possible to have access to the sdk of solexcc too? 
There is my id:  44894678

**Comment by joshbowyer** [comment]
Sorry been working on quite a few other projects so didnt want to rush, but is that deb available now?

**Comment by kellyschrock** [comment]
Hi Josh,

Sure thing, here's a link to the 64-bit Linux DEB:
https://www.dropbox.com/s/09ilz58ipmk9i9w/solex_1.0.0_amd64.deb?dl=0

Be advised, it's a pretty big archive (60-ish MB) since it's an Electron
app. Let me know how it works for you. I use it really frequently for
testing/tweaking vehicles and it's working well for me, but I haven't gone
through it and done a lot of "polishing". Some things to look out for:


   1. Video support at this point is limited to a GStreamer view external
   to Solex's main window.
   2. The "terminal" in the flight screen is more useful than it looks. In
   the flight screen, hit Ctrl+T or hit the Terminal button and then type
   "help" to get a list of commands. Type "help (command)" to get help on that
   command. I use it for a ton of stuff, and I'm interested to see if anyone
   else finds it useful!
   3. It's probably going to be a bit slow on a slow-ish laptop. Mine is a
   4-year old Lenovo X1 Extreme and it's pretty snappy on that, so you'll
   probably be fine. :-)

This has been what I'll call "thoroughly tested" on Ubuntu 20.04 and 22.04,
not sure how it'll work on other distributions. Probably fine, but I can't
say for sure.

Let me know what you think!

**Comment by joshbowyer** [comment]
Awesome thank you! I'll be testing it on Debian Bookworm so can report any missing Ubuntu-specific libs, etc

**Comment by kellyschrock** [comment]
Cool, sounds good!

**Comment by relaxibus** [comment]
Hi, any update on Solex Desktop App for Linux?

**Comment by kellyschrock** [comment]
Yes, lots of updates. I'm adding features at a pretty regular cadence, but
haven't released them yet.

**Comment by relaxibus** [comment]
any ETA? Very interested :-)

**Comment by kellyschrock** [comment]
I don't have an ETA yet, but I've added a lot more features since the last update. A  New features include:

- Button/stick mapping. Similar to what's found in Solex on the Herelink, but more extensive. It handles buttons, switches, sliders, triggers, etc on PS* and XBox controllers. You can make button mapping sets for different vehicles you use with Solex. It's also extensible. If you have some other kind of gamepad, you can write an interface for it and install it as part of a plugin. There's a library of built-in mappable actions (mode change, COMMAND_LONG, arm, disarm, etc) and you can also make your own actions for any device you might have installed on your drone that you need to control with buttons or joysticks.
- Plugin stuff is being extended. You can build a set of things (like I described above plus a bunch of other stuff like video pipelines, mission items, etc) and make a plugin that you can install into Solex. Once you restart, your plugins appear in Solex alongside the built-in things.
- Translations: Solex is being updated to make use of translations.

A lot of this stuff is being requested by a client, so I need to take some time and make a lot of this more "off the shelf" before I can come up with an ETA. But yeah, I'm doing a lot with it. :-)

**Comment by relaxibus** [comment]
Button Stick mapping is very cool as we have developed a new USB based rugged remote controller that is looking for such features :-)
Looking forward to test it!

**Comment by kellyschrock** [comment]
Something I'm hoping to get done in the next week or two is get some examples up to show how to build plugins. Someone asked me how to do that for something this weekend and I remembered I need to do that haha

**Comment by relaxibus** [comment]
Yeah, plugin is a good thing. We offer some onboard CAN, I2C and UART to ETH gateways as a sensor data collector, but the problem is how to show this non MAVLink compliant data to a GCS? A plugin will help.

**Comment by kellyschrock** [comment]
I pushed a set of plugin examples a couple of days ago here, if you want to take a look. Also let me know what kind of plugins you'd be interested in, and I'll make a note to add support for them.

**Comment by relaxibus** [comment]
we have smart DC-DC modules that outputs V/A/W in real time. Think about on a larger UxV where you have multiple Point of Loads and need to monitor each bus bar. Same for standard SBS (SMBus) based battery packs. With QGC or Mission Planner it is not possible to view all this power management data. So it would be good to send telemetry data via IP to a GCS. The protocol we use is SERaero-TM, but we can define also MQTT or something. Just keeping our discussion and idea exchanges alive.

**Comment by kellyschrock** [comment]
One thing I have is something called "SolexCC" that provides a consistent interface from a companion computer back to the GCS, but it requires that the companion computer run Node and SolexCC. There's a websocket interface that allows 2-way communication between the GCS and the companion computer. It's also not trivial to set up, so I've been thinking about ways to do the same sort of thing that are similar (it was meant mainly as an easier-to-modify version of the Solo's CC interface). What you're referring to here seems to be something where you could have a UI and backing code on the GCS that knows how to establish an IP connection to your hardware and display what it gets, which is definitely do-able.

**Comment by kellyschrock** [comment]
Here's SolexCC FYI: https://github.com/kellyschrock/solex-cc

**Comment by relaxibus** [comment]
Here are the SERaero uplink (C2) and downlink (TM) protocol specs. The TM is a serial version of the CANaerospace CAN for Aerospace open source protocol. Using the SERaero over IP would make it easy for UxV OEMs to add out of MAVLink C2 and Telemetry streams. MAVLink is just one protocol, but I see SOLEX as a generic GCS using plugins?

[SERaero_C2-16B_spec_v1.1.pdf](https://github.com/solexapp/solex-desktop-public/files/14787134/SERaero_C2-16B_spec_v1.1.pdf)
[SERaero-TM_spec_v2.1.3.pdf](https://github.com/solexapp/solex-desktop-public/files/14787135/SERaero-TM_spec_v2.1.3.pdf)

**Comment by kellyschrock** [comment]
Yeah, really it could be anything. I'm working on an integration with a camera right now that uses a completely separate Mavlink-over-UDP connection than the drone uses using a kind of mutated version of Mavlink for its messages. Making it work is just a matter of having the plugin establish the connection and exchanging data the way it needs to.

**Comment by relaxibus** [comment]
Any way to talk to you via email?

**Comment by kellyschrock** [comment]
Sure, my email is kellyschrock@gmail.com.

**Comment by mikemillerjr46** [comment]
As soon as I can afford it
